eegio/base/objects/bids/basebids.py

```diff
@@ -0,0 +1,142 @@
+
+    Only supports iEEG/scalp EEG data. We assume you store data as a BIDS with sub_<###> as the
+    identifiers. Allows easy querying of its own data structure via:
+
+    self.query.
+
+    TODO:
+    1. Add support for runs and acquisitions (https://mne.tools/mne-bids/generated/mne_bids.utils.get_entity_vals.html#mne_bids.utils.get_entity_vals)
+    2.
+
+    Attributes
+    ----------
+    datadir: os.PathLike
+        The base directory for storing data
+    modality: str
+        The modality of data we are reading. For example, EEG, MRI, CT, etc.
+
+    Notes
+    -----
+    datadir should be named and formatted according to BIDS format. For example:
+
+    datadir = <study_name>/raw/
+
+    maybe it will be:
+
+    datadir = <study_name>/derivatives/<derivative_category_name>
+
+    Examples
+    --------
+    """
+
+    def __init__(
+            self,
+            datadir: Union[os.PathLike, str],
+            modality: str,
+    ):
+        self.rootdir = datadir
+        self.datadir = datadir
+        self.modality = modality
+
+        # initialize a dictionary to store all of our paramters
+        self.parameter_dict = {}
+
+        # store parameters
+        self._store_params()
+
+        if not os.path.exists(self.description_fpath):
+            self.create_dataset_folder(self.datadir,
+                                       name="BIDS_DATASET")
+
+        if not os.path.exists(self.participants_tsv_fpath):
+            self.create_participants_files(self.participants_tsv_fpath, self.participants_json_fpath)
+
+    @property
+    def description_fpath(self):
+        return os.path.join(self.datadir, "dataset_description.json")
+
+    @property
+    def participants_tsv_fpath(self):
+        return os.path.join(self.datadir, "participants.tsv")
+
+    @property
+    def participants_json_fpath(self):
+        return os.path.join(self.datadir, "participants.json")
+
+    @property
+    def query(self):
+        return bids.BIDSLayout(self.datadir)
+
+    @property
+    def parameters(self):
+        return self.parameter_dict
+
+    @property
+    def data_types(self):
+        return mne_bids.utils.get_kinds(self.datadir)
+
+    @property
+    def subject_nums(self):
+        return mne_bids.utils.get_entity_vals(self.datadir, 'sub')
+
+    @property
+    def num_subjects(self):
+        return len(self.subject_nums)
+
+    def print_dir_tree(self):
+        basepath = os.path.join(self.datadir)
+        print_dir_tree(basepath)
+
+    def _store_params(self):
+        self.parameter_dict.update(
+            datadir=self.datadir,
+            modality=self.modality,
+        )
+
+    def create_dataset_folder(self, fpath, **dataset_description_kwargs):
+        # create dataset description
+        mne_bids.make_dataset_description(fpath, **dataset_description_kwargs)
+
+    def create_participants_files(self, tsv_fpath, json_fpath):
```

Comment:
I don't know forsure, but doesn't mne_bids.create_dataset for a subject automatically do this?

Reply:
Lmk if that is not the case tho.